avalanche/benchmarks/utils/avalanche_dataset.py

```diff
@@ -33,13 +33,8 @@
 from .dataset_definitions import ITensorDataset, ClassificationDataset, \
     IDatasetWithTargets, ISupportedClassificationDataset
 
-try:
-    from typing import List, Any, Iterable, Sequence, Union, Optional, \
-        TypeVar, Protocol, SupportsInt, Generic, Callable, Dict, Tuple, Literal
-except ImportError:
-    from typing import List, Any, Iterable, Sequence, Union, Optional, \
-        TypeVar, SupportsInt, Generic, Callable, Dict, Tuple
-    from typing_extensions import Protocol, Literal
```

Comment:
Some imports may seem useless but they are needed for type checking (not necessarily these lines), which is why the CI is currently failing.

Reply:
From the logs in the CI it seems that `BaseStrategy` was the missing import, so it should be better now
Is there a way I can run the `check_syntax` job locally to test everything is ok before pushing ?

Also I think that the unittests require too much memory to be run from my fork. I'm not sure if there is something I can do about it ?
<img width="959" alt="Screen Shot 2021-05-24 at 12 16 41 PM" src="https://user-images.githubusercontent.com/23531109/119333246-ef85d900-bc89-11eb-8cd1-d07a993a8bb7.png">

Reply:
I've never done the syntax check locally but I think it's using flake8, according to this action: https://github.com/cclauss/Find-Python-syntax-errors-action

The disk failure is unrelated to your PR. We are working to fix it ASAP.